remote/analyze_handler.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env python2
+
+import os
+import logging
+import subprocess
+import zipfile
+import redis
+import sched
+import time
+
+from enum import Enum
+
+LOGGER = logging.getLogger('ANALYZE HANDLER')
+LOGGER.setLevel(logging.INFO)
+ch = logging.StreamHandler()
+ch.setLevel(logging.INFO)
+formatter = logging.Formatter(
+    '%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+ch.setFormatter(formatter)
+LOGGER.addHandler(ch)
+
+
+class AnalyzeStatus(Enum):
+    ID_PROVIDED = 'ID_PROVIDED'
+    QUEUED = 'QUEUED'
+    ANALYZE_IN_PROGRESS = 'ANALYZE_IN_PROGRESS'
+    ANALYZE_COMPLETED = 'ANALYZE_COMPLETED'
+
+
+def preAnalyze(analyze_id, part_number):
+    LOGGER.info('Pre Analyze Step')
+
+    file_name = 'source'
+    file_extension = '.zip'
+
+    file_path = os.path.join('/workspace', analyze_id)
+
+    source_file_without_extension = file_name + '_' + str(part_number)
+    source_file_with_extension = source_file_without_extension + file_extension
+
+    zip_file_path = os.path.join(file_path, source_file_with_extension)
+
+    analyze_dir_path = os.path.join(file_path, source_file_without_extension)
+
+    with zipfile.ZipFile(zip_file_path) as zf:
+        try:
+            zf.extractall(analyze_dir_path)
+        except Exception:
+            LOGGER.error("Failed to extract all files from the ZIP.")
+
+    command_file_path = os.path.join(analyze_dir_path, 'sources-root', 'build_command')
+
+    with open(command_file_path) as build_action:
+        try:
+            build_command = build_action.readline()
+        except Exception:
+            LOGGER.error("Failed to read in build command.")
+
+    file_to_analyze_path = os.path.join(analyze_dir_path, 'sources-root', 'file_path')
+
+    with open(file_to_analyze_path) as read_in_file:
+        try:
+            read_in_file_path = read_in_file.readline()
+        except Exception:
+            LOGGER.error("Failed to read in build command.")
+
+    build_command_for_codechecker = build_command.replace(read_in_file_path, analyze_dir_path + '/sources-root' + read_in_file_path)
+
+    codechecker_path = os.path.join(os.getcwd(), "build/CodeChecker/bin/CodeChecker")
```

Comment:
CodeChecker could be an argument to the script too.

Reply:
Done